WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -248,17 +248,25 @@ def update_events
     end
   end
 
+  def get_nearby_competitions(competition)
+    nearby_competitions = competition.nearby_competitions(Competition::NEARBY_DAYS_WARNING, Competition::NEARBY_DISTANCE_KM_WARNING)[0, 10]
+    nearby_competitions.select!(&:isConfirmed?) unless current_user.can_view_hidden_competitions?
+    nearby_competitions
+  end
+
   def admin_edit
     @competition = Competition.find(params[:id])
     @competition_admin_view = true
     @competition_organizer_view = false
+    @nearby_competitions = get_nearby_competitions(@competition)
```

Comment:
I'd add a `describe 'GET #edit' do` block for this... it looks like we don't already have one.